### ISSUE TITLE
match_target should support | patterns or document what it expects

### ISSUE THREAD

**Issue by HadrienG2** [calebzulawski/multiversion#35]
Right now, match_target panics if I use an | match pattern like this...

```rust
match_target! {
    "x86_64+sse2" | "x86+sse2" => /* ... */,
    _ => /* ... */
}
```

...but that seems like the obvious way to support sse2 on all flavors of x86, for lack of a shorthand syntax to say just that.

And in any case, if that pattern is not supported, it should be documented as such.

**Comment by calebzulawski** [comment]
Good point... For now no patterns are supported, but that definitely should be.  Also, at minimum it should be a descriptive compile error and not a panic!

**Comment by calebzulawski** [comment]
Fixed in 1855d95 (and published as 0.7.1)
